pyfar/classes/filter.py

```diff
@@ -89,24 +131,29 @@ def __init__(
         """
         super().__init__()
         if coefficients is not None:
-            coefficients = atleast_3d_first_dim(coefficients)
+            coefficients = _atleast_3d_first_dim(coefficients)
         self._coefficients = coefficients
         if state is not None:
             if coefficients is None:
                 raise ValueError(
                     "Cannot set a state without filter coefficients")
-            state = atleast_3d_first_dim(state)
+            state = _atleast_3d_first_dim(state)
             self._initialized = True
         else:
             self._initialized = False
-        self._state = state
 
+        self._state = state
         self._sampling_rate = sampling_rate
-
         self._comment = comment
 
-    def initialize(self):
-        raise NotImplementedError("Abstract class method")
+    def init_state(self, state='zeros'):
+        self._state = state
+        self._initialized = True
+
+    @property
+    def coefficients(self):
+        """Coefficients of the filter"""
+        self._coefficients
```

Comment:
```suggestion
        return self._coefficients
```
I think the `_coefficients` need to be returned for the property.

Reply:
Yes, changed!
**NOTE THE FOLLOWING:** `FilterFIR` also stores a-coefficients internally, that should not be visible to the user. I thus overwrote the property for `FilterFIR` and added additional tests for that.